dstep/translator/Context.d

```diff
@@ -198,9 +219,72 @@ bool shouldBeAnonymous(Context context, Cursor cursor)
 }
 
 /**
-  * Returns true, if cursor is in the global scope.
-  */
+ * Returns true, if cursor is in the global scope.
+ */
 bool isGlobal(Cursor cursor)
 {
     return cursor.semanticParent.kind == CXCursorKind.CXCursor_TranslationUnit;
 }
+
+/**
+ * The collectTypeNames function scans the whole AST of the translation unit and produces
+ * a set of the type names in global scope.
+ *
+ * The type names are required for the parsing of C code (e.g. macro definition bodies),
+ * as C grammar isn't context free.
+ */
+
+bool[string] collectTypeNames(TranslationUnit translUnit)
```

Comment:
I found this snippet in the newsgroup as way to emulate sets. It clearly shows indentation that it's a set and not an associative array.

```d
alias Set(T) = void[0][T];

void add(T)(ref void[0][T] set, T value) {
    set[value] = (void[0]).init;
}

bool contains(T)(inout(void[0][T]) set, T value) {
    return (value in set) !is null;
}

Set!(string) a;
a.add("asd");
assert(a.contains("asd"));
a.remove("asd");
```